lib/order.rb

```diff
@@ -9,10 +13,73 @@ def initialize(id, products)
 
     def total
       # TODO: implement total
+      total = 0
+      products.each {|name, price| total += price}
+      total = total + (total* 0.075).round(2)
     end
 
     def add_product(product_name, product_price)
       # TODO: implement add_product
+      if !@products.has_key? product_name
+        @products[product_name] = product_price
+        true
+      else
+        false
+      end
+    end
+
+    def remove_product(product_name)
+      @products.delete product_name
+    end
+
+    def product_keys
+      @products.keys
+    end
+
+    def self.cost_item_hash(info_array)
+      hashy_hash = {}
+      while info_array != []
+        item = info_array.delete_at(0)
+        cost = info_array.delete_at(0)
+        hashy_hash[item] = cost
+      end
+      hashy_hash
+    end #end
+
+    #reads csv file for order
+    def self.get_csv_info
+      CSV.open(CSVFILE, 'r')
+    end
+
+    def self.all
+      all_orders = []
+      (self.get_csv_info).each do |ugly_string_line|
+        #sort the ugly string out
+        #{id_number => {item => price}}
+
+        info_array = (ugly_string_line[1]).split(/;|:/)
+
+        id_num = (ugly_string_line[0]).to_i
+        hashy_hash = self.cost_item_hash(info_array)
+
+
+        fulfillment_status = ugly_string_line[3]
```

Comment:
I'm not sure how I feel about `hashy_hash` - could you come up with a more descriptive name for this variable? Perhaps something like `parsed_products` or `product_prices`?